src/kakarot/execution_context.cairo

```diff
@@ -248,9 +233,28 @@ namespace ExecutionContext {
     // @param self The pointer to the execution context.
     // @param inc_value The value to increment the gas used with.
     // @return ExecutionContext The pointer to the updated execution context.
-    func increment_gas_used(
+    func increment_gas_used{range_check_ptr}(
```

Comment:
I've copied the behavior from the execution-specs, where they have `charge_gas` that reverts if there is not enough gas, see https://github.com/ethereum/execution-specs/blob/3fe6514f2d9d234e760d11af883a47c1263eff51/src/ethereum/shanghai/vm/gas.py#L99

I don't really like these "checked" or "safe" prefix/suffix as what they mean is never 100% clear without reading the code. And you need to read the code, well, you don't need the prefix/suffix

Reply:
Sure but increment_gas_used when reading the name of the function, I can't imagine it has the ability to revert, because the word "increment" does not carry that meaning
"charge" makes sense because it carries the meaning of paying something to someone, and if one's balance is too low, charge should fail

Reply:
ok I go for `ctx.charge_gas` then